lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -533,23 +535,27 @@ def fail_job(self, job_state):
 
     def __produce_log_file(self, job_state):
         pod_r = Pod.objects(self._pykube_api).filter(selector="app=" + job_state.job_id)
-        logs = ""
+        log_string = ""
         for pod_obj in pod_r.response['items']:
             try:
                 pod = Pod(self._pykube_api, pod_obj)
-                logs += "\n\n==== Pod " + pod.name + " log start ====\n\n"
-                logs += pod.logs(timestamps=True)
-                logs += "\n\n==== Pod " + pod.name + " log end   ===="
+                log_string += "\n\n==== Pod " + pod.name + " log start ====\n\n"
+                log_string += pod.logs(timestamps=True)
+                log_string += "\n\n==== Pod " + pod.name + " log end   ===="
             except Exception as detail:
-                log.info("Could not write pod\'s " + pod_obj['metadata']['name'] +
-                         " log file due to HTTPError " + str(detail))
+                log.info("Could not write log file for pod %s due to HTTPError %s",
+                         pod_obj['metadata']['name'], detail)
+        if isinstance(log_string, text_type):
```

Comment:
This isn't necessary, a file opened in text mode in python 2 can write bytestring and unicode strings. However as it is now it will fail on python 3, since you get a bytestring after encoding, while you need a unicode string.

Reply:
I can do this in a followup.

Reply:
Thanks @mvdbeek.  Sorry about the delay in my follow-up.